src/ui/layout/tutorial.md

```diff
@@ -13,183 +13,275 @@ diff2html: true
+
+## Add the Title section widget
+
+<?code-excerpt path-base="layout/lakes"?>
+
+In this section, you will create a `TitleSection` widget that resembles
+the following layout.
+
+{% include docs/app-figure.md
+   image="ui/layout/layout-sketch-title-block-unlabeled.svg"
+   caption="The Title section as sketch and prototype UI" %}
+
+1. Add the following code after the `MyApp` class.
+
+    <?code-excerpt "step2/lib/main.dart (titleSection)" title?>
+    ```dart
+    class TitleSection extends StatelessWidget {
+      const TitleSection({
+        super.key,
+        required this.name,
+        required this.location,
+      });
+
+      final String name;
+      final String location;
+
+      @override
+      Widget build(BuildContext context) {
+        return Padding(
+          padding: const EdgeInsets.all(32),
+          child: Row(
+            children: [
+              Expanded(
+                /*1*/
+                child: Column(
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  children: [
+                    /*2*/
+                    Padding(
+                      padding: const EdgeInsets.only(bottom: 8),
+                      child: Text(
+                        name,
+                        style: const TextStyle(
+                          fontWeight: FontWeight.bold,
+                        ),
+                      ),
+                    ),
+                    Text(
+                      location,
+                      style: TextStyle(
+                        color: Colors.grey[500],
+                      ),
+                    ),
+                  ],
                 ),
               ),
-            ),
-            Text(
-              'Kandersteg, Switzerland',
-              style: TextStyle(
-                color: Colors.grey[500],
+              /*3*/
+              Icon(
+                Icons.star,
+                color: Colors.red[500],
               ),
-            ),
-          ],
-        ),
-      ),
-      /*3*/
-      Icon(
-        Icons.star,
-        color: Colors.red[500],
-      ),
-      const Text('41'),
-    ],
-  ),
-);
-```
+              const Text('41'),
+            ],
+          ),
+        );
+      }
+    }
+    ```
 
-{:.numbered-code-notes}
- 1. Putting a `Column` inside an `Expanded` widget stretches
-    the column to use all remaining free space in the row.
-    Setting the `crossAxisAlignment` property to
-    `CrossAxisAlignment.start` positions the column at
-    the start of the row.
- 2. Putting the first row of text inside a `Container`
-    enables you to add padding. The second child in the
-    `Column`, also text, displays as grey.
- 3. The last two items in the title row are a star icon,
-    painted red, and the text "41". The entire row is in
-    a `Container` and padded along each edge by 32 pixels.
-    Add the title section to the app body like this:
+    {:.numbered-code-notes}
```

Comment:
@ericwindmill : You have no idea how often I want to add line numbers. In previous lifetimes, I would add them to anything longer than 10 lines.

Reply:
We don't have any standard functionality for this, but we could always implement something if people feel they will benefit from it. The main concern is accessibility and the ability for understanding to carry over to the explanations, so it would require a review of similar functionality out there today and any related accessibility concerns.

Generally (but not always) if a sample is so big that it needs multiple numbered explanations, it's best split up.

I'll do some research and keep this in mind as a use case to support!